### PR TITLE
[FrameworkBundle] encourage installing intl when String is available

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/String/Slugger/AsciiSlugger.php [hunk 1]
```diff
@@ -107,7 +107,7 @@ public function slug(string $string, string $separator = '-', string $locale = n
 
     private function createTransliterator(string $locale): ?\Transliterator
     {
-        if (isset($this->transliterators[$locale])) {
+        if (\array_key_exists($locale, $this->transliterators[$locale]) {
```
Comment (gharlan) [inline src/Symfony/Component/String/Slugger/AsciiSlugger.php]:
```suggestion
        if (\array_key_exists($locale, $this->transliterators) {
```

**Review by gharlan — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/String/Slugger/AsciiSlugger.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/String/Slugger/AsciiSlugger.php]: thanks

**Review by tgalopin — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

File: src/Symfony/Component/String/Slugger/AsciiSlugger.php [hunk 2]
```diff
@@ -107,7 +107,7 @@ public function slug(string $string, string $separator = '-', string $locale = n
 
     private function createTransliterator(string $locale): ?\Transliterator
     {
-        if (isset($this->transliterators[$locale])) {
+        if (\array_key_exists($locale, $this->transliterators) {
```
Comment (yceruto) [inline src/Symfony/Component/String/Slugger/AsciiSlugger.php]: missing `)`

**Review by javiereguiluz — APPROVED** [review]

**Comment by jakzal** [comment]
Unfortunately, this makes the `intl` extension required if the String component is present :(

**Comment by nicolas-grekas** [comment]
We should skip the notice if `\defined('PHPUNIT_COMPOSER_INSTALL')`, that should be enough.

**Comment by jakzal** [comment]
Won't this only fix the tests? The intl extension will still be required if the String component is present.

**Comment by nicolas-grekas** [comment]
Not with polyfills - or did you spot something that is not polyfilled?

**Comment by jakzal** [comment]
Nope, I understand now :)

**Pull request merged**